rust-runtime/inlineable/src/client_request_compression.rs

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_smithy_compression::body::compress::CompressedBody;
+use aws_smithy_compression::http::http_body_0_4_x::CompressRequest;
+use aws_smithy_compression::{CompressionAlgorithm, CompressionOptions};
+use aws_smithy_runtime_api::box_error::BoxError;
+use aws_smithy_runtime_api::client::interceptors::context::{
+    BeforeSerializationInterceptorContextRef, BeforeTransmitInterceptorContextMut,
+};
+use aws_smithy_runtime_api::client::interceptors::{Intercept, SharedInterceptor};
+use aws_smithy_runtime_api::client::orchestrator::HttpRequest;
+use aws_smithy_runtime_api::client::runtime_components::{
+    RuntimeComponents, RuntimeComponentsBuilder,
+};
+use aws_smithy_runtime_api::client::runtime_plugin::RuntimePlugin;
+use aws_smithy_types::body::SdkBody;
+use aws_smithy_types::config_bag::{ConfigBag, Layer, Storable, StoreReplace};
+use aws_smithy_types::error::operation::BuildError;
+use std::borrow::Cow;
+use std::{fmt, mem};
+
+#[derive(Debug)]
+pub(crate) struct RequestCompressionRuntimePlugin {
+    runtime_components: RuntimeComponentsBuilder,
+}
+
+impl RequestCompressionRuntimePlugin {
+    pub(crate) fn new() -> Self {
```

Comment:
I see `RequestCompressionRuntimePlugin` can support `new_1_0` when we have another `RequestCompressionInterceptor` that's `http_body_1_x` aware, which is good.